src/ssl/helper.cc

```diff
@@ -127,21 +143,23 @@ void Ssl::Helper::Submit(CrtdMessage const & message, HLPCB * callback, void * d
     SBuf rawMessage(message.compose().c_str()); // XXX: helpers cannot use SBuf
     rawMessage.append("\n", 1);
 
-    const auto pending = TheGeneratorRequests.find(rawMessage);
+    const auto key = GeneratorRequest::Key(rawMessage, ::Config.id().value);
+
+    const auto pending = TheGeneratorRequests.find(key);
     if (pending != TheGeneratorRequests.end()) {
         pending->second->emplace(callback, data);
         debugs(83, 5, "collapsed request from " << data << " onto " << *pending->second);
         return;
     }
 
-    GeneratorRequest *request = new GeneratorRequest;
-    request->query = rawMessage;
+    const auto request = new GeneratorRequest(key);
     request->emplace(callback, data);
-    TheGeneratorRequests.emplace(request->query, request);
+    TheGeneratorRequests.emplace(key, request);
     debugs(83, 5, "request from " << data << " as " << *request);
+
     // ssl_crtd becomes nil if Squid is reconfigured without SslBump or
     // certificate generation disabled in the new configuration
```

Comment:
>  there is a better solution: Instead of tagging pending requests with their configuration version, switch to a new Ssl::GeneratorRequests object 

The suggested approach is probably better (as a general solution) because we do not need copying the Config version to each helper (as we do it here in GeneratorRequests with GeneratorRequest::Key). Merging the recently added [refcounting](3bd118) should make it feasible, as you noted.

Reply:
@eduard-bagdasaryan, glad we are on the same page. 

Amos and I have discussed this PR and agreed that Option 2 above is a better (albeit small) step forward than the current PR solution. Eventually, the `::helper` class will handle (most of the optional/configurable/etc.) request collapsing tasks, but we do not have to implement those (significant) changes to fix the bug this PR is fixing.

I am marking this PR as a Draft blocked on the anticipated replacement PR. Once that replacement PR is in, we can decide what to do with the Config versioning code added this PR -- that small feature may be useful for other reconfiguration needs.